### PR TITLE
resource

### PR DESCRIPTION
**Pull request by boazco**

### CODE REVIEW

File: src/main/kotlin/com/hibob/feedback/dao/modles.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.*
+
+data class Feedback(
+    val id: UUID,
+    val creationDate: Date,
+    val department: Department,
+    val companyId: UUID,
+    val status: Status,
+    val responseIdList: List<UUID>,
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/dao/modles.kt]: i don't think that we need this. we will have a dedicated api for the responses in phase 2, let's leave this off for now

File: src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private fun buildActiveUser(requestContext: ContainerRequestContext): ActiveUser{
+        val cookie = requestContext.cookies[cookieName]?.value
+        val claims = Jwts.parserBuilder().setSigningKey(SECRET_KEY).build().parseClaimsJws(cookie).body
+        val companyId = claims["companyId"] as UUID
+        return ActiveUser(claims["employeeId"] as UUID, companyId)
+    }
+
+    @POST
+    @Path("/v1/create")
+    fun createFeedback(@RequestBody feedbackInput: FeedbackInput, requestContext: ContainerRequestContext): Response {
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: no need for the `@RequestBody` annotation

File: src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    companion object {
+        val cookieName = "Jwt"
+    }
+
+    private fun buildActiveUser(requestContext: ContainerRequestContext): ActiveUser{
+        val cookie = requestContext.cookies[cookieName]?.value
+        val claims = Jwts.parserBuilder().setSigningKey(SECRET_KEY).build().parseClaimsJws(cookie).body
+        val companyId = claims["companyId"] as UUID
+        return ActiveUser(claims["employeeId"] as UUID, companyId)
+    }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: i really like your way of thinking! let's do this in the filter, and use request headers to pass the user from the filter to hear. lemme know if that makes sense to you

Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: Yes it is
but I dont have filter yet
Should I add it now or should i finish the service and dao first?

Comment (elijulian) [reply src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: don't you have the filter from the JWT class?

i would say to remove this for now (even though i really like you direction), and in another PR let's enhance the filter class to add a ActiveUser object to the request header. then we can tweak the resource to fetch it from there. sound good?

Comment (boazco) [reply src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: great
Yes we have the filter.
should I use him and just add the active user info into the header?

Comment (elijulian) [reply src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: preferably in a separate PR

File: src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    @GET
+    @Path("/v1/feedbackId/{feedbackId}")
+    fun getFeedback(@PathParam("feedbackId") feedbackId: UUID, requestContext: ContainerRequestContext): Response {
+        val activeUser = 0
+        //send the service active user and feedbackId.
+        return Response.ok().build() //TO DO CHANGE IT to return the output from the service.
+    }
+
+    @GET
+    @Path("/v1/feedbackId/{feedbackId}/response/{responseId}")
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: if you have the response id, you won't need the feedback id.
but this is also phase two, so please leave it out

File: src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+
+@Component
+@Path("/feedback")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+class FeedbackResource(/*feedback Serrvice*/) {
+    companion object {
+        val cookieName = "Jwt"
+    }
```
Comment (elijulian) [inline src/main/kotlin/com/hibob/feedback/resource/FeedbackResource.kt]: remove, we'll get it from the filter when relevant

**Review by elijulian — COMMENTED** [review]
add a basic resource test that asserts 200

**Review by elijulian — APPROVED** [review]

**Pull request merged**